pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -40,13 +42,50 @@
     from omegaconf import DictConfig, ListConfig, OmegaConf
 
 
+if TYPE_CHECKING:
+    from torch.nn import Module
+    from torch.utils.data import DataLoader
+
+
 class TPUSpawnPlugin(DDPSpawnPlugin):
 
     def __init__(self, parallel_devices: Optional[List[int]] = None, **kwargs: Dict[str, Any]) -> None:
         super().__init__(parallel_devices, num_nodes=1, cluster_environment=None, sync_batchnorm=False)
         self.tpu_local_core_rank = 0
         self.start_method = None
 
+    @staticmethod
+    def _validate_dataloader(dataloaders: Union[List['DataLoader'], 'DataLoader']):
+        if not isinstance(dataloaders, list):
+            dataloaders = [dataloaders]
+
+        for dataloader in dataloaders:
+            if not has_len(dataloader):
+                raise MisconfigurationException(
+                    "TPUs do not currently support IterableDataset objects, the dataset must implement __len__."
+                    " HINT: You can mock the length on your dataset to bypass this MisconfigurationException."
+                )
+
+    @staticmethod
+    def _validate_patched_dataloaders(model: 'Module') -> None:
```

Comment:
Shouldnt the annotation be `LightningModule`? As we are accessing its attributes `*_dataloader`

Reply:
Good point! I've opted to instead use hasattr checks before trying to access `*_dataloader` stuff